### PR TITLE
Remove Moahker's method for creating a proper rotation matrix

### PR DESCRIPTION
**Pull request by mitiguy**
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/8148)

### CODE REVIEW

**Comment by mitiguy** [comment]
Feature review +@ericcousineau-tri 
Platform review +@sherm1 

---

Review status: 0 of 2 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/8148#-:-L6-Rcy9SdAKtkjNlwku:b-sdnxir)*

**Comment by EricCousineau-TRI** [comment]
:lgtm:, pending discussion items. Thanks!

---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---

*[math/rotation_matrix.h, line 218 at r1](https://beta.reviewable.io/reviews/robotlocomotion/drake/8148#-L6-tzSbBfAbNPK8BqS-:-L6-tzSbBfAbNPK8BqS0:b-qvjpf8) ([raw file](https://github.com/robotlocomotion/drake/blob/a470b1b7893d72426263daf2a6a07372dc49ef29/math/rotation_matrix.h#L218)):*
> ```objc
> }
> 
> /// (Deprecated), use @ref multibody::RotationMatrix::ProjectToRotationMatrix
> ```

 Can you use `DRAKE_DEPRECATED` here and above for `ProjectMatToOrthonormalMat`, and reword this to something like
```
/// This deprecated method now uses @ref multibody::RotationMatrix::ProjectToRotationMatrix for its implementation.
```

For tests that use these methods, can you either surround them with these statements, or just replace them with the original statements?
```
  // - Check deprecated function.
  #pragma GCC diagnostic push
  #pragma GCC diagnostic ignored "-Wdeprecated-declarations"
<insert call here>
#pragma GCC diagnostic pop  // pop -Wdeprecated-declarations
```

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/robotlocomotion/drake/8148#-:-L6-two0DBuG6txMLEFb:b-lte0wg)*

**Comment by sherm1** [comment]
Platform pass done, with some questions and comments. 

---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[math/rotation_matrix.h, line 220 at r1](https://beta.reviewable.io/reviews/robotlocomotion/drake/8148#-L6-zoBO0mSqU_ZUczS_:-L6-zoBO0mSqU_ZUczSa:b-xu7z48) ([raw file](https://github.com/robotlocomotion/drake/blob/a470b1b7893d72426263daf2a6a07372dc49ef29/math/rotation_matrix.h#L220)):*
> ```objc
> /// (Deprecated), use @ref multibody::RotationMatrix::ProjectToRotationMatrix
> template <typename Derived>
> Matrix3<typename Derived::Scalar> ProjectMatToRotMat(
> ```

BTW it looks like the behavior of this method has changed if det(M)<0 but I don't see any documentation about that. What happens?

---

*[math/rotation_matrix.h, line 223 at r1](https://beta.reviewable.io/reviews/robotlocomotion/drake/8148#-L6-zFY66e9AXByXWKPA:-L6-zFY66e9AXByXWKPB:b-dx3j68) ([raw file](https://github.com/robotlocomotion/drake/blob/a470b1b7893d72426263daf2a6a07372dc49ef29/math/rotation_matrix.h#L223)):*
> ```objc
>     const Eigen::MatrixBase<Derived>& M) {
>   using Scalar = typename Derived::Scalar;
>   const drake::Matrix3<Scalar> R = M;
> ```

BTW why this extra copy?

---

*[math/rotation_matrix.h, line 225 at r1](https://beta.reviewable.io/reviews/robotlocomotion/drake/8148#-L6-z-g44qcKOJGISXN4:-L6-z-g44qcKOJGISXN5:b-v7h2m9) ([raw file](https://github.com/robotlocomotion/drake/blob/a470b1b7893d72426263daf2a6a07372dc49ef29/math/rotation_matrix.h#L225)):*
> ```objc
>   const drake::Matrix3<Scalar> R = M;
>   return drake::multibody::RotationMatrix<Scalar>::ProjectToRotationMatrix(R,
>          NULL).matrix();
> ```

BTW this should be `nullptr` rather than NULL.

---

*[math/test/rotation_matrix_test.cc, line 70 at r1](https://beta.reviewable.io/reviews/robotlocomotion/drake/8148#-L6-zOT4EEhoXtZNBo8H:-L6-zOT4EEhoXtZNBo8I:bjgiifp) ([raw file](https://github.com/robotlocomotion/drake/blob/a470b1b7893d72426263daf2a6a07372dc49ef29/math/test/rotation_matrix_test.cc#L70)):*
> ```cpp
> 
>   // Non-rotation matrix in gets an orthonormal matrix out, using a full-rank
>   // matrix with det(M) < 0.
> ```

BTW this comment seems wrong now.

---

*[math/test/rotation_matrix_test.cc, line 74 at r1](https://beta.reviewable.io/reviews/robotlocomotion/drake/8148#-L6-zVaDBBmc70ogX3eV:-L6-zVaDBBmc70ogX3eW:byru5nu) ([raw file](https://github.com/robotlocomotion/drake/blob/a470b1b7893d72426263daf2a6a07372dc49ef29/math/test/rotation_matrix_test.cc#L74)):*
> ```cpp
>   M2 << 1, 2, 3, 4, 5, 6, -7, -8, -10;
>   EXPECT_GT(M2.determinant(), 0);
>   // Check that we get SO(3) with Moakher's formulation.
> ```

BTW this comment seems wrong also.

---

*[math/test/rotation_matrix_test.cc, line 75 at r1](https://beta.reviewable.io/reviews/robotlocomotion/drake/8148#-L60-9q98HiHn2HSAk78:-L60-9q98HiHn2HSAk79:b72ency) ([raw file](https://github.com/robotlocomotion/drake/blob/a470b1b7893d72426263daf2a6a07372dc49ef29/math/test/rotation_matrix_test.cc#L75)):*
> ```cpp
>   EXPECT_GT(M2.determinant(), 0);
>   // Check that we get SO(3) with Moakher's formulation.
>   Matrix3d R2_SO3 = ProjectMatToRotMat(M2);
> ```

BTW I don't see a test for behavior on a det(M)<0 matrix. What should happen?

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/robotlocomotion/drake/8148#-:-L60-GHdE4XzEbuxxkRa:b-4299sf)*

**Comment by EricCousineau-TRI** [comment]
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[math/test/rotation_matrix_test.cc, line 75 at r1](https://beta.reviewable.io/reviews/robotlocomotion/drake/8148#-L60-9q98HiHn2HSAk78:-L609IsC6RgDSk9NSZRb:b-469dfa) ([raw file](https://github.com/robotlocomotion/drake/blob/a470b1b7893d72426263daf2a6a07372dc49ef29/math/test/rotation_matrix_test.cc#L75)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

BTW I don't see a test for behavior on a det(M)<0 matrix. What should happen?
</blockquote></details>

Ooh, missed that change (glanced over sign changes)  - thanks for catching that!

From looking at `multibody/.../rotation_matrix_test.cc`, it looks like the case of `det(M) < 0` will throw an error.
Can this test be made to reflect that?

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/robotlocomotion/drake/8148)*

**Comment by mitiguy** [comment]
Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions.

---

*[math/rotation_matrix.h, line 218 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/8148#-L6-tzSbBfAbNPK8BqS-:-L6Ivx9ptw6ACbOciEr1:b5wwmkm) ([raw file](https://github.com/robotlocomotion/drake/blob/a470b1b7893d72426263daf2a6a07372dc49ef29/math/rotation_matrix.h#L218)):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

 Can you use `DRAKE_DEPRECATED` here and above for `ProjectMatToOrthonormalMat`, and reword this to something like
```
/// This deprecated method now uses @ref multibody::RotationMatrix::ProjectToRotationMatrix for its implementation.
```

For tests that use these methods, can you either surround them with these statements, or just replace them with the original statements?
```
  // - Check deprecated function.
  #pragma GCC diagnostic push
  #pragma GCC diagnostic ignored "-Wdeprecated-declarations"
<insert call here>
#pragma GCC diagnostic pop  // pop -Wdeprecated-declarations
```
</blockquote></details>

I added a TODO that this method is scheduled for removal.
I am planning TODO this soon.

---

*[math/test/rotation_matrix_test.cc, line 75 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/8148#-L60-9q98HiHn2HSAk78:-L6IzMjnKmRhxY93P7rJ:bfjpgyh) ([raw file](https://github.com/robotlocomotion/drake/blob/a470b1b7893d72426263daf2a6a07372dc49ef29/math/test/rotation_matrix_test.cc#L75)):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

Ooh, missed that change (glanced over sign changes)  - thanks for catching that!

From looking at `multibody/.../rotation_matrix_test.cc`, it looks like the case of `det(M) < 0` will throw an error.
Can this test be made to reflect that?
</blockquote></details>

This entire GTEST has been removed. 
It is repeated in its entirety 
(including a test for throwing an exception)  in:
~/drake/multibody/multibody_tree/math//rotation_matrix.h
GTEST_TEST(RotationMatrix, ProjectToRotationMatrix)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/8148)*

**Comment by sherm1** [comment]
Reviewed. One remaining issue: there is an exact zero determinant test that appears to be overly sensitive to roundoff.

---

Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[math/rotation_matrix.h, line 209 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/8148#-L6NaCMX0hGwK9pUK_tp:-L6NaCMX0hGwK9pUK_tq:b4dwvax) ([raw file](https://github.com/robotlocomotion/drake/blob/d660278dff135c8e0a6d7bc744ba76cf77a458c4/math/rotation_matrix.h#L209)):*
> ```objc
> 
> /// (Deprecated), use @ref multibody::RotationMatrix::ProjectToRotationMatrix
> // TODO(Mitiguy) Remove this method (replace calls to this method).
> ```

FYI per f2f: next step would be to remove the Drake calls to this function, and add DRAKE_DEPRECATED here so that outside code doesn't break but starts getting warnings telling them which method they _should_ be calling. Then after giving people some time to adapt, we nuke the function for good.

---

*[multibody/multibody_tree/math/test/rotation_matrix_test.cc, line 283 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/8148#-L6Nb6_BtNJv8Bcq_ZW6:-L6Nb6_BtNJv8Bcq_ZW7:b-k1bvfw) ([raw file](https://github.com/robotlocomotion/drake/blob/d660278dff135c8e0a6d7bc744ba76cf77a458c4/multibody/multibody_tree/math/test/rotation_matrix_test.cc#L283)):*
> ```cpp
> 
>   // Test a 3x3 matrix that is far from orthonormal.
>   m << 1, 0.1, 0.1, -0.2, 1.0, 0.1, 0.5, 0.6, 0.8;
> ```

BTW consider formatting these 3x3s like this:
```
m << 1,   0.1, 0.1,
    -0.2, 1.0, 0.1,
     0.5, 0.6, 0.8;
```
(much easier to read)

---

*[multibody/multibody_tree/math/test/rotation_matrix_test.cc, line 315 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/8148#-L6NdrZxTFOCGEDym3SQ:-L6NdrZxTFOCGEDym3SR:b-jviej6) ([raw file](https://github.com/robotlocomotion/drake/blob/d660278dff135c8e0a6d7bc744ba76cf77a458c4/multibody/multibody_tree/math/test/rotation_matrix_test.cc#L315)):*
> ```cpp
>   // Check that an exception is thrown if the rotation matrix is improper,
>   // meaning that the determinant of m happens to be zero.
>   m << 1, 2, 3, 4, 5, 6, 7, 8, 9;
> ```

BTW might be worth mentioning that this is singular because  m(2)=(m(1)+m(3))/2 (for either rows or columns). That was only obvious to Hongkai!

---

*[multibody/multibody_tree/math/test/rotation_matrix_test.cc, line 316 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/8148#-L6NfQnBGSVEi5_xZ4JX:-L6NfQnCEszkDYwVaFXO:b-kqu5g6) ([raw file](https://github.com/robotlocomotion/drake/blob/d660278dff135c8e0a6d7bc744ba76cf77a458c4/multibody/multibody_tree/math/test/rotation_matrix_test.cc#L316)):*
This test doesn't seem sufficient. I checked this in Octave for this matrix and got this result: 
```
>> det(m)
ans =   6.6613e-016
```
That would have passed the positive-determinant test but that seems wrong. 3*eps is just roundoff noise for a 3x3 matrix -- we should not proceed with the projection in that case.

I would think the condition we need to check for is that the given matrix is safely non-singular -- the determinant ought not to be anywhere near zero.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/8148#-:-L6NgOD4pmvf-VdUfcx7:b-s18700)*

**Comment by mitiguy** [comment]
Review status: 2 of 3 files reviewed at latest revision, 3 unresolved discussions.

---

*[multibody/multibody_tree/math/test/rotation_matrix_test.cc, line 316 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/8148#-L6NfQnBGSVEi5_xZ4JX:-L6OZ0Lg1dxCAalXHky0:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/0764f58543bff4a612e3ae6bb75f2f70ae5eeed1/multibody/multibody_tree/math/test/rotation_matrix_test.cc#L316)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

This test doesn't seem sufficient. I checked this in Octave for this matrix and got this result: 
```
>> det(m)
ans =   6.6613e-016
```
That would have passed the positive-determinant test but that seems wrong. 3*eps is just roundoff noise for a 3x3 matrix -- we should not proceed with the projection in that case.

I would think the condition we need to check for is that the given matrix is safely non-singular -- the determinant ought not to be anywhere near zero.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/8148)*

**Comment by EricCousineau-TRI** [comment]
Reviewed 1 of 3 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[math/test/rotation_matrix_test.cc, line 75 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/8148#-L60-9q98HiHn2HSAk78:-L6P_5EJn6OOxbjwJtDw:b-pps48j) ([raw file](https://github.com/robotlocomotion/drake/blob/a470b1b7893d72426263daf2a6a07372dc49ef29/math/test/rotation_matrix_test.cc#L75)):*
<details><summary><i>Previously, mitiguy wrote…</i></summary><blockquote>

This entire GTEST has been removed. 
It is repeated in its entirety 
(including a test for throwing an exception)  in:
~/drake/multibody/multibody_tree/math//rotation_matrix.h
GTEST_TEST(RotationMatrix, ProjectToRotationMatrix)


</blockquote></details>

Could you still add a test somewhere which at least calls the deprecated method, and explicitly mentions that it's deprecated (as a contract to ensure that it is not altered / removed prematurely)?

I've done something like this here:
https://github.com/RobotLocomotion/drake/blob/91df2dfbfa63ada05ab08491d6b1cba380bcd03c/multibody/test/rigid_body_tree/rigid_body_tree_creation_test.cc#L108-L113

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/8148)*

**Comment by EricCousineau-TRI** [comment]
Reviewed 1 of 3 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/8148)*

**Comment by sherm1** [comment]
Review pass done. I'm still concerned about `RotationMatrix::IsDeterminantPositive()` which I think is noise-sensitive and can be removed, PTAL.

---

Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[math/test/rotation_matrix_test.cc, line 75 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/8148#-L60-9q98HiHn2HSAk78:-L6SAd6NYXkfUtvk9Mqq:bc8aiyx) ([raw file](https://github.com/robotlocomotion/drake/blob/a470b1b7893d72426263daf2a6a07372dc49ef29/math/test/rotation_matrix_test.cc#L75)):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

Could you still add a test somewhere which at least calls the deprecated method, and explicitly mentions that it's deprecated (as a contract to ensure that it is not altered / removed prematurely)?

I've done something like this here:
https://github.com/RobotLocomotion/drake/blob/91df2dfbfa63ada05ab08491d6b1cba380bcd03c/multibody/test/rigid_body_tree/rigid_body_tree_creation_test.cc#L108-L113
</blockquote></details>

@mitiguy hasn't actually deprecated these yet since they aren't using `DRAKE_DEPRECATED` (there are still a few calls in the code base). But maybe it would be better to deprecate them for real in this PR?

---

*[multibody/multibody_tree/math/test/rotation_matrix_test.cc, line 314 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/8148#-L6SBnNOWrpjhjAgJGqk:-L6SBnNOWrpjhjAgJGql:bkab5sl) ([raw file](https://github.com/robotlocomotion/drake/blob/83acb8789bf50577ebd52a9d80c863601cc2b166/multibody/multibody_tree/math/test/rotation_matrix_test.cc#L314)):*
> ```cpp
>        0, kEpsilon, 0,
>        0, 0, kEpsilon;
>   EXPECT_TRUE(std::abs(m.determinant()) < 64 * kEpsilon * kEpsilon * kEpsilon);
> ```

BTW I don't understand this line. If the point is to check that the determinant is a small positive number why is it taking the absolute value of the determinant? I would have expected something like:
```
EXPECT_TRUE(0 < m.determinant() 
    && m.determinant() < 64 * kEpsilon * kEpsilon * kEpsilon);
```

---

*[multibody/multibody_tree/math/test/rotation_matrix_test.cc, line 327 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/8148#-L6SCsLUJ7LTF0fLjCe4:-L6SCsLUJ7LTF0fLjCe5:b1ckk0f) ([raw file](https://github.com/robotlocomotion/drake/blob/83acb8789bf50577ebd52a9d80c863601cc2b166/multibody/multibody_tree/math/test/rotation_matrix_test.cc#L327)):*
> ```cpp
>       0, 0, -kEpsilon;
>   EXPECT_TRUE(m.determinant() < 0 &&
>               std::abs(m.determinant()) < 64 * kEpsilon * kEpsilon * kEpsilon);
> ```

FYI consider 
```
EXPECT_TRUE(-64 * kEpsilon * kEpsilon * kEpsilon < m.determinant 
             && m.determinant() < 0 ).
```
which is easier to understand at a glance (IMO).

---

*[multibody/multibody_tree/math/test/rotation_matrix_test.cc, line 342 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/8148#-L6SEPma_BMGFN0t88ET:-L6SEPma_BMGFN0t88EU:bws55sq) ([raw file](https://github.com/robotlocomotion/drake/blob/83acb8789bf50577ebd52a9d80c863601cc2b166/multibody/multibody_tree/math/test/rotation_matrix_test.cc#L342)):*
> ```cpp
>   // meaning the determinant of m happens to be slightly negative (near zero).
>   // One can see this matrix is nearly singular by noticing either:
>   // row(0) + row(2) ≈ 2 * row(1)  or  col(0) + col(2) ≈ 2 * col(1).
> ```

BTW this comment seems slightly wrong now that you made the matrix nonsingular. Consider saying that the 123456789 matrix is singular for the reason you said, so the perturbed matrix is almost singular and has a determinant near -3e-13.

---

*[multibody/multibody_tree/math/test/rotation_matrix_test.cc, line 347 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/8148#-L6SE0T6cwwzeLgucGNL:-L6SE0T6cwwzeLgucGNM:b-depmcp) ([raw file](https://github.com/robotlocomotion/drake/blob/83acb8789bf50577ebd52a9d80c863601cc2b166/multibody/multibody_tree/math/test/rotation_matrix_test.cc#L347)):*
> ```cpp
>        7, 8, 9 + 400 * kEpsilon;
>   EXPECT_TRUE(m.determinant() < 0 &&
>               std::abs(m.determinant()) < 1600 * kEpsilon);
> ```

FYI consider 
```
EXPECT_TRUE(-1600 * kEpsilon < m.determinant  && m.determinant() < 0 ).
```

---

*[multibody/multibody_tree/math/test/rotation_matrix_test.cc, line 359 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/8148#-L6SFJWCF9fVyFoXvWWN:-L6SFJWCF9fVyFoXvWWO:b-ikc58p) ([raw file](https://github.com/robotlocomotion/drake/blob/83acb8789bf50577ebd52a9d80c863601cc2b166/multibody/multibody_tree/math/test/rotation_matrix_test.cc#L359)):*
> ```cpp
>        7, 8, 9 - 400 * kEpsilon;
>   EXPECT_TRUE(m.determinant() > 0 &&
>               std::abs(m.determinant()) < 1600 * kEpsilon);
> ```

BTW abs not needed here. Consider
```
EXPECT_TRUE(0 < m.determinant  && m.determinant() < 1600 * kEpsilon ).
```

---

*[multibody/multibody_tree/math/test/rotation_matrix_test.cc, line 368 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/8148#-L6SFuP5O82M-SpFLr2Z:-L6SFuP5O82M-SpFLr2_:b9y1xq9) ([raw file](https://github.com/robotlocomotion/drake/blob/83acb8789bf50577ebd52a9d80c863601cc2b166/multibody/multibody_tree/math/test/rotation_matrix_test.cc#L368)):*
> ```cpp
>        4, 5, 6,
>       -7, -8, -7;
>   EXPECT_FALSE(RotationMatrix<double>::IsDeterminantPositive(m));
> ```

I don't understand why this method still exists. If it is literally only `m.determinant() > 0` then I think you should remove it. I think it should only exist if it does something substantive that is specific to RotationMatrix objects (like it checks some noise-immune threshhold for tiny positive determinants). That's what I expected when I first saw the method so I think it is actually confusing compared to the very clear and concise `m.determinant() > 0` if it doesn't do anything nontrivial. At least it is clear in the explicit form that it is going to be extremely sensitive to roundoff noise in the determinant calculation.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/8148#-:-L6SH93z3QYmLr39Bmzo:b4buzgn)*

**Comment by mitiguy** [comment]
Review status: 2 of 4 files reviewed at latest revision, 3 unresolved discussions.

---

*[math/test/rotation_matrix_test.cc, line 75 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/8148#-L60-9q98HiHn2HSAk78:-L6TY5SoHWwyarOpj6_6:bh8yhpd) ([raw file](https://github.com/robotlocomotion/drake/blob/a470b1b7893d72426263daf2a6a07372dc49ef29/math/test/rotation_matrix_test.cc#L75)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

@mitiguy hasn't actually deprecated these yet since they aren't using `DRAKE_DEPRECATED` (there are still a few calls in the code base). But maybe it would be better to deprecate them for real in this PR?
</blockquote></details>

Will do immediate following merge of this PR.

---

*[multibody/multibody_tree/math/test/rotation_matrix_test.cc, line 342 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/8148#-L6SEPma_BMGFN0t88ET:-L6TVfl8cf2LU4N8Nq8a:bxu5g40) ([raw file](https://github.com/robotlocomotion/drake/blob/83acb8789bf50577ebd52a9d80c863601cc2b166/multibody/multibody_tree/math/test/rotation_matrix_test.cc#L342)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

BTW this comment seems slightly wrong now that you made the matrix nonsingular. Consider saying that the 123456789 matrix is singular for the reason you said, so the perturbed matrix is almost singular and has a determinant near -3e-13.
</blockquote></details>

Done  -- wording changed.
Also notice the use of the approximate equal sign (≈) 
instead of the exact equal sign (=).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/8148)*

**Comment by sherm1** [comment]
Thanks, Paul -- :lgtm:

---

Reviewed 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/8148#-:-L6TcFH8PmG1AjBvRgG7:by6dhpy)*

**Comment by EricCousineau-TRI** [comment]
Reviewed 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[math/test/rotation_matrix_test.cc, line 75 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/8148#-L60-9q98HiHn2HSAk78:-L6XkDqDTb8NDEhDXwku:b-suxhpk) ([raw file](https://github.com/robotlocomotion/drake/blob/a470b1b7893d72426263daf2a6a07372dc49ef29/math/test/rotation_matrix_test.cc#L75)):*
<details><summary><i>Previously, mitiguy wrote…</i></summary><blockquote>

Will do immediate following merge of this PR.
</blockquote></details>

OK Sounds good!
(BTW, I wasn't asking that the `DRAKE_DEPRECATED` decorator is necessary in this PR, just that something in the unittest consume this function.)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/8148)*
